### PR TITLE
APP-627 Remove Dependency on Cloud-Logging

### PR DESCRIPTION
**Pull request by robson-sanchez**
- APP-722 Configure Log4j to use a custom appender
- APP-723 Move IntegrationBridgeKeyProvider to a non-open source repo
- APP-724 Link dynamically the IntegrationBridgeSessionProvider at runtime

### CODE REVIEW

File: integration-web/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -7,6 +7,9 @@
         <Property name="log4j2.symphonyLogLevel">INFO</Property>
         <Property name="log4j2.consoleLogThreshold">INFO</Property>
         <Property name="log4j2.outputAllToConsole">false</Property>
+        <Property name="symphony.instanceId">integration-bridge</Property>
+        <Property name="symphony.maxLogsPerBulkRequest">10</Property>
+        <Property name="symphony.maxPayloadSize">100000</Property>
```
Comment (ecarrenho-daitan) [inline integration-web/src/main/resources/log4j2.xml:12]: We had to put this configuration in place due to issue https://perzoinc.atlassian.net/browse/CP-1023. We have a ticket to track this and adjust if necessary in the future: https://perzoinc.atlassian.net/browse/APP-754.

**Review by ecarrenho-daitan — APPROVED** [review]
LGTM.

@dnsymphony please review.

File: integration-web/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -40,18 +43,30 @@
...
         </RollingFile>
 
+        <SymphonyOnPremHttpsAppender name="OnPremHttpsAppender" url="${sys:cloudLogHarvesterURL}"
+                                     maxLogsPerBulkRequest="${sys:symphony.maxLogsPerBulkRequest}"
+                                     maxLogsPerBulkRequestFromSink="${sys:symphony.maxLogsPerBulkRequest}"
+                                     maxPayloadSize="${sys:symphony.maxPayloadSize}" flushingInterval="5"
+                                     sinkFlushingInterval="5"
+                                     sinkFileName="${sys:logs.basedir}/logs/integration." instanceId="${sys:symphony.instanceId}"
+                                     sessionProviderClassName="com.symphony.integration.logging.IntegrationBridgeSessionProvider">
+            <PatternLayout pattern="%d %-5p [%c] (%t) %X{X-Trace-Id} %m%n"/>
```
Comment (dnsymphony) [inline integration-web/src/main/resources/log4j2.xml]: Put something around the trace ID so it's not free-floating.  Maybe '[Trace ID: %X{X-Trace-Id}]'

File: integration-web/src/test/resources/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<Configuration>
+
+    <Appenders>
+        <Console name="console" target="SYSTEM_OUT">
+            <ThresholdFilter level="INFO" onMatch="ACCEPT" onMismatch="DENY"/>
+            <PatternLayout pattern="%d{ISO8601} %-5p [%t] %20c{1} %M %X{X-Trace-Id} - %m%n"/>
```
Comment (dnsymphony) [inline integration-web/src/test/resources/log4j2.xml:7]: Same comment as above

**Review by dnsymphony — CHANGES_REQUESTED** [review]

File: integration-web/src/main/resources/log4j2.xml (hunk 2 above)
Comment (robson-sanchez) [reply integration-web/src/main/resources/log4j2.xml]: FIXED

File: integration-web/src/main/resources/log4j2.xml [hunk 4]
```diff
@@ -17,14 +17,14 @@
         <Console name="console" target="SYSTEM_OUT">
             <ThresholdFilter level="${sys:log4j2.consoleLogThreshold}" onMatch="ACCEPT"
                              onMismatch="DENY"/>
-            <PatternLayout pattern="%d{ISO8601} %-5p [%t] %20c{1} %M %X{X-Trace-Id} - %m%n"/>
+            <PatternLayout pattern="%d{ISO8601} %-5p [%t] %20c{1} %M Trace ID: %X{X-Trace-Id} - %m%n"/>
```
Comment (dnsymphony) [inline integration-web/src/main/resources/log4j2.xml]: And square brackets '[Trace ID: kdsfljsdfl]'

**Review by dnsymphony — CHANGES_REQUESTED** [review]

File: integration-web/src/test/resources/log4j2.xml (hunk 3 above)
Comment (robson-sanchez) [reply integration-web/src/test/resources/log4j2.xml:7]: FIXED

File: integration-web/src/main/resources/log4j2.xml (hunk 4 above)
Comment (robson-sanchez) [reply integration-web/src/main/resources/log4j2.xml]: FIXED

**Review by dnsymphony — APPROVED** [review]

**Pull request merged**